utils/vlog/tests/v2x_tests.cmake

```diff
@@ -35,19 +35,25 @@ function(V2X_TEST_GENERIC)
   # pb_type checking
   set(GOLDEN_XML golden.${TYPE}.xml)
   add_file_target(FILE ${GOLDEN_XML} SCANNER_TYPE xml)
-  xml_canonicalize_merge(NAME ${NAME}_${TYPE}_golden_sort FILE ${GOLDEN_XML} OUTPUT ${NAME}.${TYPE}.golden.xml)
+  xml_canonicalize_merge(NAME ${NAME}_${TYPE} FILE ${GOLDEN_XML} OUTPUT ${NAME}.${TYPE}.golden.xml)
 
   set(ACTUAL_XML ${NAME}.${TYPE}.xml)
-  xml_canonicalize_merge(NAME ${NAME}_${TYPE}_actual_sort FILE ${ACTUAL_XML} OUTPUT ${NAME}.${TYPE}.actual.xml)
+  xml_canonicalize_merge(NAME ${NAME}_${TYPE} FILE ${ACTUAL_XML} OUTPUT ${NAME}.${TYPE}.actual.xml)
 
-  diff(NAME ${NAME}_${TYPE}_diff GOLDEN ${NAME}.${TYPE}.golden.xml ACTUAL ${NAME}.${TYPE}.actual.xml)
+  get_rel_target(REL_DIFF_NAME diff ${NAME}.${TYPE}.xml)
+  diff(NAME ${REL_DIFF_NAME} GOLDEN ${NAME}.${TYPE}.golden.xml ACTUAL ${NAME}.${TYPE}.actual.xml)
 
-  add_dependencies(all_v2x_tests ${NAME}_${TYPE}_diff)
+  add_dependencies(all_v2x_tests ${REL_DIFF_NAME})
+
+  set(FILE_DEPS "")
+  append_file_dependency(FILE_DEPS ${ACTUAL_XML})
 
   # update golden model/pb_type target
-  add_custom_target(${NAME}_update_golden_${TYPE})
+  get_rel_target(REL_GOLDEN_XML update ${GOLDEN_XML})
+  add_custom_target(${REL_GOLDEN_XML})
   add_custom_command(
-    TARGET ${NAME}_update_golden_${TYPE}
+    TARGET ${REL_GOLDEN_XML}
+    DEPENDS ${FILE_DEPS}
```

Comment:
If `ACTUAL_XML` is already present (even if it is changed by v2x), does the update golden command get executed anyway?

Reply:
This was extracted into https://github.com/SymbiFlow/symbiflow-arch-defs/pull/728